extensions/ql-vscode/src/data-extensions-editor/extension-pack-picker.ts

```diff
@@ -303,12 +308,145 @@ async function pickNewExtensionPack(
   }
 
   const name = matches.groups.name;
-  const packPath = join(workspaceFolder.path, name);
+  const packPath = join(workspaceFolder.uri.fsPath, name);
 
   if (await pathExists(packPath)) {
     return undefined;
   }
 
+  return writeExtensionPack(packPath, packName, databaseItem.language);
+}
+
+async function autoCreateExtensionPack(
+  name: string,
+  language: string,
+  extensionPacksInfo: QlpacksInfo,
+  logger: NotificationLogger,
+): Promise<ExtensionPack | undefined> {
+  const workspaceFolder = await autoPickWorkspaceFolder(language);
+  if (!workspaceFolder) {
+    return undefined;
+  }
+
+  const packName = autoNameExtensionPack(name, language);
+  if (!packName) {
+    void showAndLogErrorMessage(
+      logger,
+      `Could not automatically name extension pack for database ${name}`,
+    );
+
+    return undefined;
+  }
+
+  const existingExtensionPackPaths = extensionPacksInfo[packName];
+  if (existingExtensionPackPaths?.length === 1) {
+    let extensionPack: ExtensionPack;
+    try {
+      extensionPack = await readExtensionPack(existingExtensionPackPaths[0]);
+    } catch (e: unknown) {
+      void showAndLogErrorMessage(
+        logger,
+        `Could not read extension pack ${packName}`,
+        {
+          fullMessage: `Could not read extension pack ${packName} at ${
+            existingExtensionPackPaths[0]
+          }: ${getErrorMessage(e)}`,
+        },
+      );
+
+      return undefined;
+    }
+
+    return extensionPack;
+  } else if (existingExtensionPackPaths?.length > 1) {
+    void showAndLogErrorMessage(
+      logger,
+      `Extension pack ${packName} resolves to multiple paths`,
+      {
+        fullMessage: `Extension pack ${packName} resolves to multiple paths: ${existingExtensionPackPaths.join(
+          ", ",
+        )}`,
+      },
+    );
+
+    return undefined;
+  }
+
+  const matches = packNameRegex.exec(packName);
```

Comment:
Why is this check here? If we autogenerated the name, shouldnt it match the regex?

Reply:
I've removed it and made the `autoNameExtensionPack` a bit more robust and testable so that it should not be possible for the name to not match the regex.